core/spec.md

```diff
@@ -595,38 +599,45 @@ the display name.
 
 Note, since `SINGULARid` is immutable, in order to change its value a new
 entity would need to be created with the new `SINGULARid` that is a deep-copy
-of the existing entity. Then the existing entity can be deleted.
+of the existing entity. Then the existing entity would be deleted.
 
 ##### `self` Attribute
 
 - Type: URL
-- Description: A server generated unique URL for the current entity. In the
-  case of pointing to an entity in a [Registry
-  Collection](#registry-collections), the URL MUST be a combination of the
-  base URL for the collection appended with the `SINGULARid` of the entity.
-
-  When this URL references a Resource or Version, and the Resource has the
-  `hasdocument` model aspect set to `true`, and `?doc` mode is not enabled,
-  then it MUST include `$details` appended to its `SINGULARid` if the request
-  asked for the serialization of the xRegistry metadata. This would happen
-  when `$details` was used in the request, or when the Resource (or Version)
-  is included in the serialization of a parent entity. If `?doc` mode
-  is enabled, then `$details` MUST appear when this URL is an absolute URL,
-  otherwise for relative URLs it MUST NOT appear.
-
-  If the Resource or Version has the `hasdocument` model aspect set to `false`,
-  then this URL MAY be appended with `$details` only if the request was
-  directed at the Resource or Version and included `$details`. Otherwise in
-  all other situations `$details` MUST NOT be included.
+- Description: A server generated unique URL referencing the current entity.
+  - Each entity in the Registry MUST have a unique `self` URL value that
+    locates where in the Registry hierarchy the entity exists and from where
```

Comment:
```suggestion
    locates the entity in the Registry hierarchy and from where
```
I think this reads a bit easier, feel free to ignore if you disagree

Reply:
Done